graylog/resource/user/create.go

```diff
@@ -15,16 +15,30 @@ func create(d *schema.ResourceData, m interface{}) error {
 		return err
 	}
 	data, err := getDataFromResourceData(d)
+	if cl.APIVersion == "v4" {
+		delete(data, "full_name")
+	}
 	if err != nil {
 		return err
 	}
 	if _, ok := data[keyPermissions]; !ok {
 		data[keyPermissions] = []string{}
 	}
 
-	if _, err := cl.User.Create(ctx, data); err != nil {
+	_, err = cl.User.Create(ctx, data)
+	if err != nil {
 		return err
 	}
-	d.SetId(data[keyUsername].(string))
+	if cl.APIVersion == "v4" {
+		new_data, _, _ := cl.User.Get(ctx, data[keyUsername].(string), "v3")
```

Comment:
Should the last parameter `v3` be `v4`, as we are within the `v4` IF block?

Reply:
It's due to the strangeness of the graylog api which does not send back the id on creation

Reply:
👍